src/imitation/rewards/reward_net.py

```diff
@@ -49,101 +50,156 @@ def __init__(
+        # a reward network. Going to keep this for compat for now, but I want
+        # to remove these methods in the future and push all the tensor/ndarray
+        # conversion and preprocessing logic back up the call stack.
+        return self._eval_reward(self._reward_train, state, action, next_state, done)
+
+    def reward_test(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Compute the test reward with raw ndarrays, including preprocessing.
 
         Note this is the reward we use for transfer learning.
 
-        Different concrete subclasses will require different
-        placeholders to be filled to calculate this output, but for all
-        subclasses, filling the following placeholders will be sufficient:
-
-        ```
-        self.obs_ph
-        self.act_ph
-        self.next_obs_ph
-        ```
+        Args:
+          state: current state.
+          action: action associated with `state`.
+          next_state: next state.
+          done: 0/1 value indicating whether episode terminates on transition
+            to `next_state`.
 
         Returns:
-          tf.Tensor: A (None,) shaped Tensor holding
-            the test reward associated with each timestep.
+          np.ndarray: A (None,) shaped ndarray holding the test reward
+            associated with each timestep.
         """
-        return self._theta_output
+        return self._eval_reward(self._reward_test, state, action, next_state, done)
+
+    def _eval_reward(
+        self,
+        method: Callable[[th.Tensor, th.Tensor, th.Tensor, th.Tensor], th.Tensor],
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Evaluates either train or test reward, given appropriate method."""
+        # torchify
+        dev = self.device()
+        state_th = th.as_tensor(state, device=dev)
+        action_th = th.as_tensor(action, device=dev)
+        next_state_th = th.as_tensor(next_state, device=dev)
+        done_th = th.as_tensor(done, device=dev)
+
+        # preprocess observations/actions
+        state_th = preprocess_obs(state_th, self.observation_space, self.scale)
+        action_th = preprocess_obs(action_th, self.action_space, self.scale)
+        next_state_th = preprocess_obs(
+            next_state_th, self.observation_space, self.scale
+        )
+        done_th = done_th.to(th.float32)
+
+        with th.no_grad():
+            th_reward = self._reward_test(state_th, action_th, next_state_th, done_th)
+
+        rew = th_reward.detach().cpu().numpy().flatten()
+        assert rew.shape == state.shape[:1]
+        return rew
 
     @abstractmethod
-    def build_theta_network(
-        self, inputs: Sequence[tf.Tensor]
-    ) -> Tuple[tf.Tensor, networks.LayersDict]:
+    def build_base_reward_network(self) -> nn.Module:
```

Comment:
Meh, easier to have base reward net as an abstract method instead of forcing it to be constructed here.